woocommerce/woocommerce-ios
# Screenshots: Fix errors generating promo screenshots with Fastlane

Related: pdnsEh-hU-p2
⚠️ Depends on fixes from https://github.com/wordpress-mobile/release-toolkit/pull/366 — `/fastlane/Pluginfile` will need to be updated to point the official `4.2.0` release with that fix, once that release is out. ⚠️ 

## Description

To generate a set of screenshots for the App Store, we start with raw screenshots from the app and compose them into promo screenshots (with the device frame, caption, and background) with the Fastlane `create_promo_screenshots` lane. This PR resolves some errors that currently pop up when generating the promo screenshots with Fastlane.

Note: There is one unresolved issue with the captions on the promo screenshots in `en_US`. This is related to the `promo_screenshot_*.txt` file names used for those captions; it will be addressed after other tooling changes are complete, per the discussion in https://github.com/woocommerce/woocommerce-ios/pull/6936#discussion_r881790345.

## Changes

* Updates `Fastfile` to call `has_git_lfs?`, resolving an undefined method error.
* Updates the Fastlane `Pluginfile` to install rmagick version 4.1. This matches the version of rmagick used in the release toolkit, which contains the action and helper used to actually compose the promo screenshots. This update resolves an error that appears if an earlier version of rmagick is installed.
* Updates the Fastlane `Pluginfile` to point to the `trunk` branch of the release toolkit, to pick up the fixes to the promo screenshots helper from https://github.com/wordpress-mobile/release-toolkit/pull/366. (As noted above, this will eventually need to be updated to point to the release toolkit `4.2.0` release.)
* Updates `screenshots.json` to use the correct raw screenshot filenames to match the changes (reordering) done in https://github.com/woocommerce/woocommerce-ios/pull/5936.
* Updates several Italian screenshot caption strings with the latest translations from GlotPress.

## Testing

1. Get a full set of raw screenshots from here: p91TBi-8mI-p2#comment-10218.*
2. Run `bundle install --with screenshots` to install `rmagick`.
3. If needed, run `brew install automattic/build-tools/drawText` to install `drawText`.
4. Run `bundle exec fastlane create_promo_screenshots` to generate the promo screenshots.
5. Confirm a full set of promo screenshots is created in each locale in `/fastlane/screenshots`.

*The raw screenshots are generated with `bundle exec fastlane screenshots`. However, that lane is not working right now, due to a breaking change made after the [last set of fixes](https://github.com/woocommerce/woocommerce-ios/pull/6699). I'll follow up with a fix for that in a separate PR.

## Submitter Checklist

Update release notes:
